algorithms/src/sorting/impl/Kokkos_SortImpl.hpp

```diff
@@ -268,19 +268,29 @@ void copy_to_host_run_stdsort_copy_back(
     KE::copy(exec, view, view_dc);
 
     // run sort on the mirror of view_dc
-    auto mv_h  = create_mirror_view_and_copy(Kokkos::HostSpace(), view_dc);
-    auto first = KE::begin(mv_h);
-    auto last  = KE::end(mv_h);
-    std::sort(first, last, std::forward<MaybeComparator>(maybeComparator)...);
+    auto mv_h = create_mirror_view_and_copy(Kokkos::HostSpace(), view_dc);
+    if (view.span_is_contiguous()) {
+      std::sort(mv_h.data(), mv_h.data() + mv_h.size(),
+                std::forward<MaybeComparator>(maybeComparator)...);
+    } else {
```

Comment:
Please comment (here in reply) what this added complexity brings to the test?

Reply:
What test do you mean?

Reply:
Did not read it right. Sorry.